Add TypeScript docs (#220)
civiform/docs: docs/contributor-guide/developer-guide/testing.md

#### De-randomizing

Timestamp/Dates and applicant IDs will change each time a test is run, to
automatically normalize these, UI elements that contain them need to have the
[`ReferenceClasses`](https://sourcegraph.com/github.com/civiform/civiform/-/blob/server/app/views/style/ReferenceClasses.java?L133&subtree=true)

> `BT_DATE`, `BT_APPLICATION_ID` and few other classes added respectively.

Reviewer: ```suggestion
`BT_DATE`, `BT_APPLICATION_ID` and a few others added respectively.
```

Author: Done.

> Check [normalizeElements](https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/civiform/civiform%24+normalizeElements&patternType=standard&sm=1)

Reviewer: ```suggestion
Check the [normalizeElements](https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/civiform/civiform%24+normalizeElements&patternType=standard&sm=1)
```

Author: Done.

> function for up-to-date list of elements and update if necessary.

Reviewer: ```suggestion
function for an up-to-date list of elements and update if necessary.
```
